google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemTest.java

```diff
@@ -334,6 +337,39 @@ public void testDeleteRecursive() throws IOException {
     }
   }
 
+  @Test
+  public void testSameProvider() throws IOException {
+    try (CloudStorageFileSystem sourceFileSystem =
+        CloudStorageFileSystem.forBucket(
+            "bucket",
+            CloudStorageConfiguration.builder().permitEmptyPathComponents(true).build())) {
+      CloudStorageFileSystem destFileSystem =
+          CloudStorageFileSystem.forBucket(
+              "new-bucket",
+              CloudStorageConfiguration.builder().permitEmptyPathComponents(true).build());
+      assertEquals(sourceFileSystem.provider(), destFileSystem.provider());
```

Comment:
I think you can replace `assertEquals` with `assertSame` here

Reply:
Done